### DISCUSSION TITLE
useRouter() from "next/router" vs "next/navigation"

### DISCUSSION THREAD

**Discussion by shlomo20** [vercel/next.js#41928] [Help]
Hi, 
can someone explain the difference between the two useRouter();?

consider  a page like this 

```javascript
export default function SignIn({ providers  }: ISignInProps) {
  const { data: session, status } = useSession()
  const router = useRouter()  
  if(session) {
    if(session.user?.type === 'biz') router.push('/businessPortal')
    else{
      router.push('/unauthorized')
    } 
  }else{  
       return (<div > hello form signin page</div>)
   }
```

now if i import the useRouter from "next/router" this would work fine however if i import useRouter from "next/navigation" 
it will result in the folwing Error 

> 
> Unhandled Runtime Error
> TypeError: Cannot read properties of null (reading **'push')**

**Comment by raajnadar** [comment]
Did you find the documentation for the `next/navigation`?

**Reply by shlomo20** [reply]
Yes, however there is nothing there would explain the difference in behavior or with what it's different.
 https://beta.nextjs.org/docs/routing/linking-and-navigating

**Reply by raajnadar** [reply]
Oh it is under beta, I was searching the next 13 stable document

**Comment by arielvieira** [comment]
useRouter from `next/router` includes `router.query`, and `router.pathname` when the useRouter from `next/navigation` does not, if you want the query and pathname, you should use `useSearchParams` and `usePathname` from `next/navigation`.

**Comment by apecollector** [comment]
How can I get the query parameters and path name from the server? 

They used to be in context in next 12 `getServerSideProps`.

I ask because using this in a server component:

```
  const pathName = usePathname();
  const searchParams = useSearchParams();

  console.log(pathName);
  console.log(searchParams);
```

gives an error:

```
Error: Cannot read properties of null (reading 'useContext')
```
